packages/components/src/settings/index.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { Form } from '../index';
+import {
+	SettingText,
+	SettingPassword,
+	SettingCheckbox,
+	SettingSelect,
+} from './types';
+
+const typeMap = {
+	text: SettingText,
+	password: SettingPassword,
+	checkbox: SettingCheckbox,
+	select: SettingSelect,
+};
+
+export const Settings = ( {
```

Comment:
This component could be called `SettingsForm`, what do you think?

Reply:
Sound good to me 👍🏻